cluster/images/barbican-kms-plugin/Dockerfile

```diff
@@ -10,23 +10,67 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG ALPINE_ARCH=amd64
-# We not using scratch because we need to keep the basic image information
-# from parent image
-FROM ${ALPINE_ARCH}/alpine:3.15.4
+################################################################################
+##                               BUILD ARGS                                   ##
+################################################################################
+# This build arg allows the specification of a custom Golang image.
+ARG GOLANG_IMAGE=golang:1.19.6
 
-ARG ARCH=amd64
+# We use Alpine as the source for default CA certificates
+ARG ALPINE_IMAGE=alpine:3.15.4
+
+ARG COMPONENT=barbican-kms-plugin
+
+################################################################################
+##                              BUILD STAGE                                   ##
+################################################################################
+# Fetch ca-certificates from Alpine
+FROM ${ALPINE_IMAGE} as certs
+RUN apk add --no-cache ca-certificates
+
+# Build the manager as a statically compiled binary so it has no dependencies
+# libc, muscl, etc.
+FROM --platform=linux/amd64 ${GOLANG_IMAGE} as builder
+
+ARG GOPROXY=https://goproxy.io,direct
+ARG TARGETOS
+ARG TARGETARCH
+ARG VERSION
+ARG COMPONENT
+
+WORKDIR /build
+COPY go.mod go.sum ./
+COPY cmd/ cmd/
+COPY pkg/ pkg/
+RUN GO111MODULE=on CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} GOPROXY=${GOPROXY} \
+                go build \
+                -trimpath \
+                -ldflags="-w -s -X k8s.io/component-base/version.gitVersion=${VERSION}" \
+                -o=$COMPONENT \
+                cmd/$COMPONENT/main.go
+
+################################################################################
+##                               MAIN STAGE                                   ##
+################################################################################
+
+# barbican-kms-plugin uses ALPINE instead of distroless because its entrypoint
+# uses a shell for environment substitution. If there are no other uses this
+# could be replaced by callers passing arguments explicitly.
+
+FROM --platform=${TARGETPLATFORM} ${ALPINE_IMAGE}
+
+ARG COMPONENT
+
+COPY --from=builder /build/$COMPONENT /bin/$COMPONENT
+COPY --from=certs /etc/ssl/certs /etc/ssl/certs
```

Comment:
I compared previous code a little bit ,just curious ,the reason of copy those certs is because alpine doen'st have those certs so we have to copy for ssl connection?

Reply:
Actually I originally wrote it for the distroless occm image. I kept it the same for all images just to keep it the same, tbh. I was thinking of a potential situation where Debian and Alpine CA are briefly out of sync and occm can connect to OpenStack but cinder-csi can't. I figured they could all have exactly the same ca-certs from the same source copied in the same way.